config/suppressions.xml

```diff
@@ -179,4 +179,10 @@
   <!-- DetailAST API requires large number of methods -->
   <suppress checks="MethodCount"
             files="[\\/]src[\\/]main[\\/]java[\\/]com[\\/]puppycrawl[\\/]tools[\\/]checkstyle[\\/]DetailAstImpl.java"/>
+  <!-- setFileContents is used within the class, it's not a parameter -->
+  <suppress id="settersHaveSinceTag"
+            files="SuppressionCommentFilter|SuppressWithNearbyCommentFilter"/>
```

Comment:
this a good modules with properties, please share what is problem with them ?

Reply:
`fileContents` is not a property here. It is a setter used internally
https://checkstyle.org/filters/suppressioncommentfilter.html#Properties

Reply:
https://github.com/checkstyle/checkstyle/blob/63b4549b0c51924e266af2d60730911bc6358b86/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java#L586-L595

Reply:
can we use some nearby filters to suppress only this setter , not whole file ?

Reply:
please do:
```
 $ git diff
diff --git a/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java b/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
index 74d247e..8e23b8e 100644
--- a/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java
@@ -587,10 +587,8 @@ public class SuppressionCommentFilter
      * Set the FileContents for this filter.
      *
      * @param fileContents the FileContents for this filter.
-     * @noinspection WeakerAccess
-     * @noinspectionreason WeakerAccess - we avoid 'protected' when possible
      */
-    public void setFileContents(FileContents fileContents) {
+    private void setFileContents(FileContents fileContents) {
         fileContentsReference = new WeakReference<>(fileContents);
     }
 

```

Reply:
Done.